From: Debian Deep Learning Team <debian-ai@lists.debian.org>
Date: Sun, 6 Feb 2022 12:19:35 +0100
Subject: Use system GTest

---
 CMakeLists.txt       |  1 +
 tests/CMakeLists.txt | 21 ++++++++++++++-------
 2 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7b1fc6..14109c3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,7 @@ option(FAISS_OPT_LEVEL "" "generic")
 option(FAISS_ENABLE_GPU "Enable support for GPU indexes." ON)
 option(FAISS_ENABLE_PYTHON "Build Python extension." ON)
 option(FAISS_ENABLE_C_API "Build C API." OFF)
+option(FAISS_USE_SYSTEM_GTEST "Use system-provided gtest library." ON)
 
 if(FAISS_ENABLE_GPU)
   set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index b0bdfd0..5d8ae42 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -30,18 +30,25 @@ else()
   target_link_libraries(faiss_test PRIVATE faiss)
 endif()
 
-include(FetchContent)
-FetchContent_Declare(googletest
-  URL "https://github.com/google/googletest/archive/release-1.10.0.tar.gz")
-set(BUILD_GMOCK CACHE BOOL OFF)
-set(INSTALL_GTEST CACHE BOOL OFF)
-FetchContent_MakeAvailable(googletest)
+if(NOT FAISS_USE_SYSTEM_GTEST)
+  include(FetchContent)
+  FetchContent_Declare(googletest
+    URL "https://github.com/google/googletest/archive/release-1.10.0.tar.gz")
+  set(BUILD_GMOCK CACHE BOOL OFF)
+  set(INSTALL_GTEST CACHE BOOL OFF)
+  FetchContent_MakeAvailable(googletest)
+else()
+  find_package(GTest REQUIRED)
+  include_directories(${GTEST_INCLUDE_DIRS})
+endif()
+
 
 find_package(OpenMP REQUIRED)
 
 target_link_libraries(faiss_test PRIVATE
   OpenMP::OpenMP_CXX
-  gtest_main
+  ${GTEST_BOTH_LIBRARIES}
+  pthread
 )
 
 # Defines `gtest_discover_tests()`.
