--- faiss.orig/faiss/CMakeLists.txt
+++ faiss/faiss/CMakeLists.txt
@@ -275,14 +275,20 @@ target_include_directories(faiss_avx512
 set_target_properties(faiss PROPERTIES
   POSITION_INDEPENDENT_CODE ON
   WINDOWS_EXPORT_ALL_SYMBOLS ON
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
 )
 set_target_properties(faiss_avx2 PROPERTIES
   POSITION_INDEPENDENT_CODE ON
   WINDOWS_EXPORT_ALL_SYMBOLS ON
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
 )
 set_target_properties(faiss_avx512 PROPERTIES
   POSITION_INDEPENDENT_CODE ON
   WINDOWS_EXPORT_ALL_SYMBOLS ON
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
 )
 
 if(WIN32)
--- a/c_api/CMakeLists.txt
+++ b/c_api/CMakeLists.txt
@@ -5,7 +5,11 @@
 
 cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
 
-project(faiss_c_library LANGUAGES C CXX)
+project(faiss_c_library 
+  VERSION 1.8.0
+  DESCRIPTION "C bindings for faiss."
+  HOMEPAGE_URL "https://github.com/facebookresearch/faiss"
+  LANGUAGES C CXX)
 
 set(CMAKE_C_STANDARD 11)
 
@@ -34,6 +38,19 @@
 add_library(faiss_c ${FAISS_C_SRC})
 target_link_libraries(faiss_c PRIVATE faiss)
 
+set_target_properties(faiss_c PROPERTIES
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+install(TARGETS faiss_c
+  EXPORT faiss-targets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
 function(faiss_install_headers headers p)
   foreach(h ${headers})
     get_filename_component(f ${h} DIRECTORY)
@@ -51,6 +68,16 @@
 
 faiss_install_headers("${FAISS_C_API_HEADERS}" c_api)
 
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/libfaiss_c.pc.in
+  ${CMAKE_CURRENT_BINARY_DIR}/libfaiss_c.pc
+  @ONLY
+)
+install(
+  FILES ${CMAKE_CURRENT_BINARY_DIR}/libfaiss_c.pc
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
+)
+
 add_executable(example_c EXCLUDE_FROM_ALL example_c.c)
 target_link_libraries(example_c PRIVATE faiss_c)
 
